user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleBatchInformationSpreadsheet.java

```diff
@@ -0,0 +1,384 @@
+        .toList();
+
+    sampleCodeColumn = addColumn("Sample code",
+        sampleInfo -> sampleInfo.getSampleCode().orElse(""),
+        SampleInfo::setSampleCode)
+        .requireDistinctValues();
+    lockColumn(sampleCodeColumn);
+    hideColumn(sampleCodeColumn);
+
+    addColumn("Analysis to be performed",
+        SampleInfo::getAnalysisToBePerformed,
+        AnalysisMethod::label,
+        (sampleInfo, analysisToBePerformed) -> sampleInfo.setAnalysisToBePerformed(
+            AnalysisMethod.forLabel(analysisToBePerformed)))
+        .selectFrom(sortedAnalysisMethods,
+            identity(),
+            getAnalysisMethodItemRenderer())
+        .setRequired();
+
+    addColumn("Sample label", SampleInfo::getSampleLabel,
+        SampleInfo::setSampleLabel)
+        .requireDistinctValues()
+        .setRequired();
+
+    addColumn("Condition", SampleInfo::getExperimentalGroup,
+        experimentalGroup -> formatConditionString(experimentalGroup.condition()),
+        (sampleInfo, conditionString) -> updateSampleInfoWithMatchingExperimentalGroup(
+            experimentalGroups, sampleInfo, conditionString))
+        .selectFrom(experimentalGroups, identity())
+        .setRequired();
+
+    addColumn("Biological replicate ID",
+        SampleInfo::getBiologicalReplicate,
+        BiologicalReplicate::label,
+        (sampleInfo1, value) -> updateSampleInfoWithMatchingBiologicalReplicate(sampleInfo1, value,
+            experimentalGroups))
+        .selectFrom(sampleInfo -> Optional.ofNullable(sampleInfo.getExperimentalGroup())
+            .map(ExperimentalGroup::biologicalReplicates).orElse(List.of()), identity())
+        .setRequired();
+
+    addColumn("Species",
+        SampleInfo::getSpecies,
+        Species::label,
+        SampleInfo::setSpecies)
+        .selectFrom(this.species, identity())
+        .setRequired();
+
+    addColumn("Specimen",
+        SampleInfo::getSpecimen,
+        Specimen::label,
+        SampleInfo::setSpecimen)
+        .selectFrom(this.specimens, identity())
+        .setRequired();
+
+    addColumn("Analyte",
+        sampleInfo -> Optional.ofNullable(sampleInfo.getAnalyte())
+            .map(Analyte::label)
+            .orElse(null),
+        SampleInfo::setAnalyte)
+        .selectFrom(this.analytes, Analyte::label)
+        .setRequired();
+
+    addColumn("Customer comment", SampleInfo::getCustomerComment,
+        SampleInfo::setCustomerComment);
+
+    for (int i = 0; i < INITIAL_ROW_COUNT; i++) {
+      addEmptyRow();
+    }
+  }
+
+  public void showSampleCode() {
+    showColumn(sampleCodeColumn);
+  }
+
+  public void hideSampleCode() {
+    hideColumn(sampleCodeColumn);
+  }
+
+  /**
+   * adds an empty row to the spreadsheet
+   */
+  public void addEmptyRow() {
+    ValidationMode validationMode = this.validationMode;
+    setValidationMode(ValidationMode.LAZY);
+    addRow(new SampleInfo());
+    setValidationMode(validationMode);
+  }
```

Comment:
MAJOR NIT: 
I'm unsure if this is a default behaviour since it's not specfic concerning the sampebatchinformation. Could be moved into an interface Vaadin style 🤔